app/components/Flow.tsx

```diff
@@ -28,6 +42,16 @@ export function Flow() {
           className='bg-slate-300 rounded hover:bg-slate-400 transition-all'
         />
       </ReactFlowProvider>
+
+      <Modal
+        isOpen={isModalOpen}
+        onClose={closeModal}
+        nodeData={
+          selectedNode && typeof selectedNode.data.label === 'string'
+            ? { id: selectedNode.id, label: selectedNode.data.label }
+            : null
+        }
```

Comment:
this could be a `const` created in the component body, to make it more readable.

Reply:
it really gets better, I'm going to change